packages/stokhos/src/sacado/kokkos/vector/Stokhos_MP_Vector_MaskTraits.hpp

```diff
@@ -228,145 +263,152 @@ template<typename S>  Sacado::MP::Vector<S> operator* (Sacado::MP::Vector<S> &a1
     Sacado::MP::Vector<S> mul;
     mul = m*a1;
     return mul;
-};
+}
 
 template<typename S, typename S2>  Sacado::MP::Vector<S> operator* (S2 a1, Mask<Sacado::MP::Vector<S>> m)
 {
     Sacado::MP::Vector<S> mul;
     mul = m* (Sacado::MP::Vector<S>) a1;
     return mul;
-};
+}
 
 template<typename S, typename S2>  Sacado::MP::Vector<S> operator* (Mask<Sacado::MP::Vector<S>> m, S2 a1)
 {
     Sacado::MP::Vector<S> mul;
     mul = m* (Sacado::MP::Vector<S>) a1;
     return mul;
-};
+}
 
+template<typename S> Mask<Sacado::MP::Vector<S> > signbit_v (Sacado::MP::Vector<S> &a1)
+{
+    typedef EnsembleTraits_m<Sacado::MP::Vector<S> > ET;
+    typedef typename ET::value_type value_type;
+    
+    Mask<Sacado::MP::Vector<S> > mask;
+    for(int i=0; i<ET::size; ++i)
+        mask[i] = std::signbit<value_type>(ET::coeff(a1,i));
```

Comment:
I think the source of the compiler failure is specifying the template parameter in the signbit function.  It should be just
```
mask[i] = std::signbit(ET::coeff(a1,i));
```
because the `double` version is an overload, and not a template function.  By including the template parameter, you are selecting the integral one, which the `std::enable_if` fails on.

Reply:
It is strange, it worked on my laptop. I will correct that.